src/JustSaying/Fluent/ServiceResolver/ServiceBuilderServiceResolver.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using JustSaying.AwsTools;
+using JustSaying.Messaging.MessageHandling;
+using JustSaying.Messaging.MessageSerialization;
+using JustSaying.Messaging.Monitoring;
+using Microsoft.Extensions.Logging;
+
+namespace JustSaying.Fluent
+{
+    internal class ServiceBuilderServiceResolver : IServiceResolver
+    {
+        private readonly ServicesBuilder _builder;
+
+        private readonly ConcurrentDictionary<Type, object> _serviceLookup = new();
+
+        private bool _built = false;
+
+        public ServiceBuilderServiceResolver(ServicesBuilder builder)
+        {
+            _builder = builder;
+
+        }
+
+        public T ResolveService<T>() where T : class
+        {
+            throw new NotImplementedException();
```

Comment:
It was intentional as this service resolver doesn't support optional resolutions, but instead of throwing an `InvalidOperationException` here it's probably cleaner to just implement ResolveService like this. Thoughts?
```csharp
public T ResolveService<T>() where T : class
{
    return ResolveOptionalService<T>() ??
        throw new InvalidOperationException(
            $"Service type {typeof(T).FullName} isn't available from this service resolver.");
}
```